### ISSUE TITLE
Bug: Playlist modification actions not working in light mode in V1.0.9

### ISSUE THREAD

**Issue by pkbott** [mpogue2/SquareDesk#1149]
Tried adding a playlist in light mode and was not able to at all, no adding/creating/saving/etc.  

Switched to dark mode and could add to a playlist column but could not sort the music at all and could only add more to the bottom of the playlist.  I have never really used this feature so maybe it's partly me.  Went looking for documentation on it in the help section and there isn't much there for me to go on.  

Interesting that after creating a playlist in dark mode that it would only work in dark mode too.  I have quite a few songs in the playlist in dark mode and when switching to light mode, there is only one song in the playlist.  

Needless to say there is a bunch not working here in light mode.

**Comment by mpogue2** [comment]
Hmm...I will check it out in light mode...

In dark mode, you can drag to a slot, and it goes in at the bottom. We don't have dragging working yet within a list.  Use SHIFT CMD up/down/left/right to move them up/down/to top/to bottom...

**Comment by pkbott** [comment]
Test loop button missing in light mode now tooReady to try something FUN, UNIQUE and ***@***.*** Square Dance Group:  https://www.facebook.com/groups/squaredanceforfun/On Jul 25, 2024, at 6:51 PM, Mike Pogue ***@***.***> wrote:﻿
Hmm...I will check it out in light mode...
In dark mode, you can drag to a slot, and it goes in at the bottom. We don't have dragging working yet within a list.  Use SHIFT CMD up/down/left/right to move them up/down/to top/to bottom...

**Comment by mpogue2** [comment]
@pkbott OK, here's what I found on these two issues:

- In Light Mode, I can LOAD and SAVE playlists just fine, but I can't add songs to it, delete songs from it, move songs up/down, etc. even from the right-click context menu.  Good catch! Definitely a bug.  :-/

- In Light mode, there wasn't ever a Test Loop button on the Music tab.  Light Mode just has the menu item Music > Test Loop, and its shortcut CMD-SHIFT-L....  Adding a button for Test Loop was done for Dark Mode, once the Music tab had copies of all of the loop controls added to it.

**Comment by mpogue2** [comment]
How weird.  I had it failing every time, and now it doesn't fail ever in Light Mode.
I added a Track filter in Dark Mode, and now it fails again in Light Mode.  This is strange.  I have a theory as to what's happening...

**Comment by mpogue2** [comment]
My theory was incorrect.

I've gotten it to fail only once more in Light Mode.  I'm not sure if it's an intermittent failure, or if something I'm doing is triggering it (rarely).

**Comment by mpogue2** [comment]
OK, I MIGHT have a reproducible test case now.
Wow this is weird.

- Start up 1.0.9 in Light Mode.
- Clear Playlist
- in the songTable, choose one item and Add to Playlist (it adds)
- Load a Recent playlist
- Select 2nd item in the new playlist, and try to move it (doesn't move)

**Comment by mpogue2** [comment]
Yes, a test on just PlaylistItemsToTop() suggests that this fixes the context menu problem.
It does NOT fix the keyboard shortcuts problem, which appears to be separate.

**Comment by mpogue2** [comment]
Note that while Dark Mode playlists can have the same song listed more than once in the playlists, Light Mode playlists cannot.  They will only register the LAST song of that name, and the first instance of that song will be missing a number in the # column.

**Comment by mpogue2** [comment]
OK, I needed to fix a problem in the globalEventFilter, too (was this bug there all along?).  It wasn't letting CMD-SHIFT-LEFT/RIGHT through to the playlist handling functions, when the focus was in one of the three filter text fields in Light Mode.

I think that's all fixed now, and I tested:
- dark mode playlist keyboard shortcuts OK
- dark mode playlist right-click context menu OK
- light mode playlist keyboard shortcuts OK
- light mode playlist right-click context menu OK
- light mode playlist menu OK

Commit: cdc0c7bc966a8a2709cd78e5c8e51e0a38020ea1

**Comment by mpogue2** [comment]
@pkbott I'm gonna turn this comment "Interesting that after creating a playlist in dark mode that it would only work in dark mode too. I have quite a few songs in the playlist in dark mode and when switching to light mode, there is only one song in the playlist." into a new bug, because I don't know how to reproduce this, and I think it's a separate problem.

**Comment by pkbott** [comment]
Ok, I know that there was a test loop button using light mode on the cue sheet tab since I used that exclusively setting up my patter music but it is now gone.  Was this recently changed somehow?  I even looked in 1.0.5 and it doesn't show up there either and I know that it did.

**Comment by mpogue2** [comment]
@pkbott The Test Loop button is there in all versions in all Light Modes (I went back all the way to 0.9.5), but it's only visible (in all of these versions) when a Patter song is loaded.  It's intentionally not visible for singing calls.

So, if you have no song loaded, or if you have a song loaded that is not Patter, then you won't see the button!

If this is the case, but you still need the button (e.g. you have a folder name that contains Patter but is not recognized as such by SquareDesk, e.g. perhaps a singing call is used as Patter), then maybe the algorithm should be different:

"if (Patter) then Test Loop Button = visible" is what we have today, but perhaps

a) "if (not Singing Call) then Test Loop Button = visible", or perhaps even
b) Test Loop Button always visible

might be better.

**Comment by pkbott** [comment]
no patter test loop button showing and a patter is loaded

<img width="1440" alt="Screenshot 2024-07-30 at 7 20 35 AM" src="https://github.com/user-attachments/assets/77ac572f-b822-489a-b3db-71d1a1213bcb">

**Comment by mpogue2** [comment]
@pkbott  Cool, thanks!  Is this song not in the usual "Patter" folder in the Music Directory?

In your case, if the song title ends in "(Patter)", regardless of which folder it's in (even if a singing call folder), do you  consider it to be Patter (like an "override" of the folder that it's in, maybe?)?

Or maybe option c) above is the best approach to handle this (i.e. just leave the button visible all the time)?

In this example below from my setup, the Music tab of Preferences shows both the Music Directory path, and the Music Types (of which "patter" is one).

So, everything in 
- `/Users/mpogue/Library/CloudStorage/Box-Box/__squareDanceMusic_Box/patter` and
- `/Users/mpogue/Library/CloudStorage/Box-Box/__squareDanceMusic_Box/hoedown`
will be considered to be patter (and anything NOT in those 2 directories does not show the Test Loop button when loaded).  Loading any song in these 2 directories should make the Test Loop button visible on the Cuesheet page in Light Mode.

![image](https://github.com/user-attachments/assets/c4c0397b-a23f-4ec0-ae56-03a9adeff685)

**Comment by mpogue2** [comment]
@pkbott I think your other comment belongs attached to this one (re: Test Loop Button)...so, copying it here

"This is my fault. I totally forgot. I took and  changed the name of the
patter folder on purpose to get the colors to flip between dark and light
mode automatically for the text that’s why there’s no more test loop button"

**Comment by pkbott** [comment]
I personally like the test loop for all but that is me.  I do use singing
calls as both patter and singers. Currently to make them loop, I was
copying them into both patter and singing call folders so the lyrics would
show and that I could set up the loop all while in LIGHT mode.
